### ISSUE TITLE
Missing accessibility in Progress component

### ISSUE THREAD

**Issue by marysehirtz** [chakra-ui/chakra-ui#6899]
### Description

Update Progress so it doesn't cause an accessibility violation.

### Problem Statement/Justification

It causes problems with screen readers as reported by the storybook accessibility plugin. This is the error message you get when you use the Progress component:
![image](https://user-images.githubusercontent.com/116345122/197513148-fb863455-1c1b-45e2-8aef-d91c2725e234.png)
Adding a title doesn't seem to have any impact on the error message since the title is added to the outer div and not the progressbar itself.

Adding an aria-label leads to this error message:
![image](https://user-images.githubusercontent.com/116345122/197528171-bc0929e8-628c-46c9-890b-3a9f0e36070d.png)

### Proposed Solution or API

Maybe add the title to the inner div of the progressbar?
`<div class="css-1pd7wb5">`
`<div  title="progress" aria-valuemax="5" aria-valuemin="0" aria-valuenow="1" role="progressbar" class="css-4bdnun" style="width: 20%;"></div>`
`</div>`
